File: 05_arrays/05_16_maximum_subarray/MaximumSubarray.java
import java.util.Arrays;

public class MaximumSubarray {
    public static void main(String[] args) {
        int[] arr = { -2, 1, -3, 4, -1, 2, 1, -5, 4 };
        System.out.println("Before sorting:");
        System.out.println(Arrays.toString(arr));
        int result = maximumSubarray(arr);
        System.out.println("The maximum subarray sum is: " + result);
    }

    private static Integer maximumSubarray(int[] arr) {
        int result = Integer.MIN_VALUE;
        int sum = 0;

        for (int i : arr) {
            sum += i;
            result = Math.max(result, sum);
            if (sum < 0) {
                sum = 0;
            }
        }

        return result;
    }
}

File: 05_arrays/05_16_maximum_subarray/maximum_subarray.js
const maximumSubarray = (arr = []) => {
    let result = Number.MIN_VALUE;
    let sum = 0;

    for (let i = 0; i < arr.length; i++) {
        sum += arr[i];
        result = Math.max(result, sum);
        sum = sum < 0 ? 0 : sum; 
    }

    return result;
}


let arr = [-2, 1, -3, 4, -1, 2, 1, -5, 4];
console.log("Before sorting:");
console.log(arr.join(" "));
let result = maximumSubarray(arr);
console.log("The maximum subarray sum is:", result);